File: python/util/detab.py
def detab ( filename ):
   """detab( 'filename' )

   Turns all tabs in a file in to spaces.
   """

   f1 = file( filename, 'r' )
   text = f1.read()
   f1.close()

   text = text.split( '\t' )
   sep = ' ' * 8
   text = sep.join( text )

   f1 = file( filename, 'w' )
   f1.write( text )
   f1.close()


def main():
   import sys
   if (len( sys.argv ) < 2 ):
      print "Usage:  detab.py filename"
      sys.exit( 0 )
   else:
      detab( sys.argv[1] )
      print "File", sys.argv[1], "de-tabbed"

if __name__ == '__main__':
   main()

File: c/bin/detab.c
/* detab.c
 * turns tabs into three-space indents.
 * 
 * Simon Heath
 * 3/11/2002
 */

#include <stdio.h>
#include <string.h>

void usage()
{
   printf( "Turns all the tabs in a file into spaces.\n" );
   printf( "Usage:  detab <infile> <outfile>\n\n" );
   exit( 0 );
}


main( int argc, char **argv )
{
   FILE *infile, *outfile;
   int ch;

   if( argc < 2 )
   {
      usage();
   }

   infile = fopen( argv[1], "r" );
   outfile = fopen( argv[2], "a" );

   while( (ch = getc( infile )) != EOF )
   {
      if( ch == '\t' )
         fputs( "        ", outfile );
      else
         putc( ch, outfile );
   }

   fclose( infile );
   fclose( outfile );
}


